Library/Homebrew/extend/pathname.rb

```diff
@@ -216,10 +216,10 @@ def cp_path_sub(pattern, replacement)
 
   # extended to support common double extensions
   def extname(path = to_s)
-    BOTTLE_EXTNAME_RX.match(path)
-    return $1 if $1
-    /(\.(tar|cpio|pax)\.(gz|bz2|lz|xz|Z))$/.match(path)
-    return $1 if $1
+    bottle_ext = path[BOTTLE_EXTNAME_RX, 1]
+    return bottle_ext if bottle_ext
+    archive_ext = path[/(\.(tar|cpio|pax)\.(gz|bz2|lz|xz|Z))$/, 1]
```

Comment:
Did you intentionally drop the `.dmg` and `.zip` extensions again? They were present in the Cask version and in your original version of this PR, but maybe they are indeed no longer needed.